src/lib/colorUtils.ts

```diff
@@ -9,3 +17,43 @@ export const mapColorsToScheme = <T>(
     { ...siteVars.emphasisColors, ...siteVars.naturalColors },
     typeof mapper === 'number' ? String(mapper) : (mapper as any),
   ) as ColorValues<T>
+
+export const getColorSchemeFn = <T>(colorProp: string, colorScheme: ColorValues<T>) => {
+  const colors = _.get(colorScheme, colorProp)
+  return (area: keyof T, defaultColor: string) => (colors ? colors[area] : defaultColor)
+}
+
+export const getColorSchemeFromObject = (
+  colorScheme: ColorValues<Partial<ColorScheme>>,
+  colors: ComplexColorPropType,
+): Partial<ColorScheme> =>
+  _.mapValues(colors, (color, colorName) => {
+    // if the color scheme contains the color, then get the value from it, otherwise return the color provided
+    const colorSchemeValue = _.get(colorScheme, color, colorScheme.default[color])
+    return colorSchemeValue ? colorSchemeValue[colorName] : colors[colorName]
+  })
+
+export const getColorSchemeWithCustomDefaults = (
+  colorScheme: ColorSchemeMapping,
+  customDefaultValues: Partial<ColorScheme>,
+) => {
+  const mergedDefaultValues = {
+    ...colorScheme.default,
+    ...customDefaultValues,
+  }
+  return {
+    ...colorScheme,
+    default: mergedDefaultValues,
+  }
+}
+
+export const generateColorScheme = (
+  colorProp: ComplexColorPropType,
+  colorScheme: ColorValues<Partial<ColorScheme>>,
+): Partial<ColorScheme> => {
+  return _.isObject(colorProp)
+    ? { ...colorScheme.default, ...getColorSchemeFromObject(colorScheme, colorProp) }
+    : _.get(colorScheme, colorProp as string, {
+        ...colorScheme.default,
+      })
```

Comment:
1. We don't need destruction there
```js
_.get(colorScheme, colorProp as string, colorScheme.default)
```
2. If we will revert the condition, we will be able to omit `as string`
```
type of colorProp === 'string'
    ? _.get(colorScheme, colorProp, colorScheme.default)
    : { ...colorScheme.default, ...getColorSchemeFromObject(colorScheme, colorProp) }
```